librz/util/str.c

```diff
@@ -753,22 +753,26 @@ RZ_API char *rz_str_newf(const char *fmt, ...) {
 	return p;
 }
 
-// Secure string copy with null terminator (like strlcpy or strscpy but ours
+/**
+ * \brief Secure string copy with null terminator
+ *
+ * 	This API behaves like strlcpy or strscpy.
+ */
 RZ_API size_t rz_str_ncpy(char *dst, const char *src, size_t n) {
 	rz_return_val_if_fail(dst && src, 0);
-	size_t i;
 
 	// do not do anything if n is 0
 	if (n == 0) {
 		return 0;
 	}
+#if HAVE_STRLCPY
+	return strlcpy(dst, src, n);
+#else
+	strncpy(dst, src, n);
+	dst[n] = '\0';
```

Comment:
This is probably where the issue is @officialcjunior . strlcpy (and rz_str_ncpy) takes the full size as argument, including the null byte. However you usually have to pass n-1 to strncpy and of course dst[n] goes out of bound because the last char is dst[n-1].